java/app/src/main/java/com/fullstorydev/shoppedemo/adapters/CheckoutBindingAdapter.java

```diff
@@ -1,12 +1,14 @@
 package com.fullstorydev.shoppedemo.adapters;
 
-import android.widget.Spinner;
+import android.widget.AutoCompleteTextView;
 
 import androidx.databinding.BindingAdapter;
 
 public class CheckoutBindingAdapter {
     @BindingAdapter({"selectedValue"})
-    public static void setSelectedValue (Spinner s, int position) {
-        s.setSelection(position);
+    public static void setSelectedValue (AutoCompleteTextView s, int position) {
+        String text = String.valueOf(s.getAdapter().getItem(position));
+        // set filter to false: disable the filtering and always show full list
+        s.setText(text,false);
```

Comment:
we have to disable filtering behavior for `auto complete`, if enabled, auto fill will filter out the options, unless user manually delete the auto fill text